### PR TITLE
[FEATURE] Registers a student to the platform

### LINKED ISSUE #4: [FEATURE] Registers a student in the platform
**Is your feature request related to a problem? Please describe.**
A student has to register to be enrolled in the platform
**Describe the solution you'd like**
Students are required to provide their personal details, like full name, level of education, age.
**Describe alternatives you've considered**
A student can be asked to provide additional details like their weak subjects and favorite subjects, this information will be used to suggest the best courses for him/her.

**Additional context**
None.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### PR DESCRIPTION
**Pull request by Njaya2019**
## Description
It adds a student to the platform.

### client request
```
POST users/students/register
```
### request body
```
{
    "firstname": "Weclif",
    "middlename": "Mzae",
    "lastname": "Kisao",
    "Address": "Mombasa, Kenya",
    "email": "willytuvaK0313@gmail.com",
    "age": 13,
    "educationlevel": "Olevel"
}
```
### successful response example
```
{
    "id": 1,
    "firstname": "Willy",
    "middlename": "Mzae",
    "lastname": "Kisao",
    "Address": "Mombasa, Kenya",
    "email": "willymzae@gmail.com",
    "age": 13,
    "educationlevel": "Olevel"
}
```
### successful response type and code
```
 response type: application/json
 status code: 201
```
### Dependencies
```
- Django==3.0.6
- djangorestframework==3.11.0
- pylint==2.5.2
- coverage==4.4
- pytest==5.4.2
- pytest-cov==2.8.1
- pytest-django==3.9.0
- pytest-env==0.6.2
```
**Note** tests won't run if *coverage* is a latest version
All pull requests should be related to open issues. Indicate the issue(s) bellow and remove this line:
Fixes #(issue number #4 )

## How Has This Been Tested?
**Clone**
```
git clone --single-branch --branch ft-registers-a-student  git@github.com:BuildForSDG/team-195.git
```
**Installation**
change directory to the tutor folder
```
team-195> cd tutor
```
Install all dependencies first
```
pip install -r requirements/local.txt 
pip install -r requirements/base.txt 
```
Set environment variables for both test and base database.
```
DATABASE_URL=psql://USER:PASSWORD@HOST:HOST/your-database
TEST_DATABASE=psql://USER:PASSWORD@HOST:HOST/test-database
```
Run migrations
```
- python manage.py makemigrations
- python manage.py migrate
```
run
```
python manage.py runserver
```

## Screenshots (if applicable, else remove this line / section)
**Bad request image**
image
**Successful request image**
image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [x] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by BernardMW** [comment]
I think this is perfect. However, I would like us to make use of the sign up model that we get by default. What you have very good. But it can still be improved.

**Comment by BernardMW** [comment]
I particularly applaud your work on serializers. Great job.

**Review by BernardMW — APPROVED** [review]

**Review by BernardMW — DISMISSED** [review]

**Review by BernardWambua — DISMISSED** [review]
This is okay. However I think if you can inherit from the user registration model it will even be better.

**Review by peterkuria — DISMISSED** [review]
Great progress. The API endpoint is working well. I incorporated my own feedback in your pull request with a few additional commits. Test passed after reviewing the few bugs such as timezone.

**Comment by BernardWambua** [comment]
I think Education level field can be a lookup field.

**Comment by BernardWambua** [comment]
@Njaya2019 Kindly change the education level to lookup. It's the only issue I see with this branch the we can merge it.

**Review by BernardWambua — APPROVED** [review]
Approved.

File: tutor/config/settings/base.py [hunk 1]
```diff
@@ -2,21 +2,24 @@
...
 env = environ.Env()
 
-READ_DOT_ENV_FILE = env.bool("DJANGO_READ_DOT_ENV_FILE", default=False)
-if READ_DOT_ENV_FILE:
-    # OS environment variables take precedence over variables from .env
-    env.read_env(str(ROOT_DIR / ".env"))
+READ_DOT_ENV_FILE = env("DJANGO_READ_DOT_ENV_FILE", default=False)
+
+# OS environment variables take precedence over variables from .env
+env.read_env(str(ROOT_DIR / ".env"))
```
Comment (peterkuria) [inline tutor/config/settings/base.py:22]: changed environmental variable clause..

**Review by peterkuria — APPROVED** [review]
Approved changes to db config

**Pull request merged**